internal/k8s/controllers/config_controller.go

```diff
@@ -114,6 +119,10 @@ func (r *ConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		Nodes:              nodes.Items,
 	}
 
+	for _, node := range nodes.Items {
+		r.Nodes[node.Name] = node
```

Comment:
I think every round must be deleted and recreated otherwise we won't be able to deal with deletions (or, you must iterate over it and remove nodes not in items, but recreating seems enough)

Reply:
correct, but anyway this local cache is no longer needed with new predicate version which overrides `UpdateFunc`.